builder/setup-builder.nix

```diff
@@ -3,28 +3,48 @@
 { setup-depends, package, name, src, flags }:
 
 let
+  component = {
```

Comment:
There is a similar setup of a dummy component in the `shellFor` function. Is the same change not needed there?

Reply:
I've had a look and `shellFor` does not pass in the source, so it does not need to call `cleanCabalComponent` with the component.  The extra fields are only used by `cleanCabalComponent`.